src/components/displaySvg/displaySvg.js

```diff
@@ -1,48 +1,73 @@
-import React, { Component } from 'react';
-import base64 from 'base-64';
-import PropTypes from 'prop-types';
+import React, { Component } from "react";
+import base64 from "base-64";
+import "./finalSvgDisplay.scss";
+import PropTypes from "prop-types";
+import InlineSVG from "svg-inline-react";
 class DisplaySvg extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            fileSize: 0
-        };
-    }
-
-    componentWillMount = () => {
-        if (this.props.dataUrl) {
-            this.setState({
-                fileSize: this.dataURLtoKilloBytes(this.props.dataUrl.split(',')[1])
-            });
-        }
+  constructor(props) {
+    super(props);
+    this.state = {
+      fileSizeNormal: 0,
+      fileSizeOptimised: 0,
+      isClicked: false
     };
+  }
 
-    dataURLtoKilloBytes = dataURI => {
-        return base64.decode(dataURI).length / 1000;
-    };
+  changeColourOnclickToMark = () => {
+    this.setState({ isClicked: !this.state.isClicked });
+  };
 
-    render() {
-        return (
-            <div>
+  componentWillMount = () => {
+    if (this.props.dataUrl) {
+      this.setState({
+        fileSizeNormal: this.dataURLtoKilloBytes(
+          this.props.dataUrl.split(",")[1]
+        ),
+        fileSizeOptimised: this.props.finalStringElementWrappedWithLink.length / 1000
+      });
+    }
+  };
+
+  dataURLtoKilloBytes = dataURI => {
+    return base64.decode(dataURI).length / 1000;
+  };
+
+  render() {
+    return (
+      <div>
+        <div className="display-svgs-border-match">
+            <ul>
+              <li>
                 <figure>
-                    <img
-                        id='svg-display-id'
-                        src={this.props.dataUrl}
-                        width={this.props.width}
-                        height={this.props.height}
-                        alt='svg to be displayed'
-                    />
-                    <figcaption>{ 'File Size : '+this.state.fileSize +' KB'}</figcaption>
+                  <img
+                    className={this.props.svgType}
+                    src={this.props.dataUrl}
+                    width={this.props.width}
+                    height={this.props.height}
+                    alt="svg to be displayed"
+                  />
+                  <figcaption>
+                    {"File Size : " + this.state.fileSizeNormal + " KB"}
+                  </figcaption>
                 </figure>
-            </div>
-        );
-    }
+              </li>
+              <li>
+                <InlineSVG src={this.props.finalStringElementWrappedWithLink} />
```

Comment:
null check, anywhere where you have a this.props, use a nullcheck

Reply:
I have handled the case where a null maybe passed here.

Reply:
Null checks have been done.